src/modules/currency/wallet/currency-wallet-callbacks.js

```diff
@@ -40,6 +40,64 @@ export function forEachListener (
   }
 }
 
+type ThrottleDetails = {
+  [walletId: string]: {
+    delayCallback: boolean,
+    lastCallbackTime: number,
+    txArray: Array<EdgeTransaction>
+  }
+}
+
+const throttleDetailsNewTx: ThrottleDetails = {}
+const throttleDetailsChangedTx: ThrottleDetails = {}
+
+let throttleRateLimitMs = 5000
+
+function throttledTxCallback (
+  throttleDetails: ThrottleDetails,
+  callback: Function,
+  walletId: string,
+  txArray: Array<EdgeTransaction>
+) {
+  // If this is a unit test, lower throttling to 250ms
+  if (txArray[0].txid.length < 3) {
+    throttleRateLimitMs = 250
+  }
+  if (!throttleDetails[walletId]) {
+    throttleDetails[walletId] = {
+      delayCallback: false,
+      lastCallbackTime: 0,
+      txArray: []
+    }
+  }
+  if (!throttleDetails[walletId].delayCallback) {
+    const now = Date.now()
+    if (
+      now - throttleDetails[walletId].lastCallbackTime >
+      throttleRateLimitMs
+    ) {
+      callback(walletId, txArray)
+      throttleDetails[walletId].txArray = []
+      throttleDetails[walletId].lastCallbackTime = Date.now()
+    } else {
+      console.log(
+        'throttledTxCallback setTimeout delay callback id:' + walletId
+      )
```

Comment:
I would like to see `io.console.log` instead of `console.log` Since the core is intended to be used in other projects, we have to be more well-behaved than the GUI, which owns the world. I know the GUI just uses `console.log` everywhere, but the core is still 100% using the io object.